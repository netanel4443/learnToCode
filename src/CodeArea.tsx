import React, { useEffect } from 'react'
import { View, Text, ScrollView, StyleSheet, } from 'react-native'
import SyntaxHighlighter from 'react-native-syntax-highlighter'; 
import { dark} from 'react-syntax-highlighter/styles/hljs';
import HTMLView from 'react-native-htmlview';
import {SPLIT_CHAR} from './articles/index'
import intents from './articles/intent'
import alertDialog from './articles/alertDialog'
import activityLifecycle from './articles/activityLifecycle'

const splitBetweenTextAndCode=()=>{
  const splitedArr=activityLifecycle.split(SPLIT_CHAR)
  const elementsArr:any[]=[]
  splitedArr.forEach((textOrCode,index)=>{
  
   if(index%2!==0){
    elementsArr.push(
      <SyntaxHighlighter
        style={dark}
        customStyle={{marginLeft:3,marginRight:3,borderRadius:3}}
        language='kotlin'
        fontSize={15}
        highlighter={"hljs"}
        wrapLongLines
      >
        {textOrCode}
      </SyntaxHighlighter>
    )
   }
   else{
    elementsArr.push(
      <View style={{margin:10}}>
        <HTMLView
          value={`<p>${textOrCode}</p>`}
          stylesheet={htmlStyles}
        />
      </View>
    )
   }
  })
  return elementsArr
}

const CodeArea = () => {
 
  return (
    <View style={{flex:1,backgroundColor:'#e6e6e6'}}>
      <ScrollView>
       {splitBetweenTextAndCode()}
      </ScrollView>
    </View>
    
  )
}

const htmlStyles=StyleSheet.create({
  p:{
    fontSize:15,
    fontFamily:'serif',
    lineHeight:25
  }
})

export default CodeArea
