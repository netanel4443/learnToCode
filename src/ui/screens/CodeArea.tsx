import React, { useCallback, useEffect } from 'react'
import { View,  ScrollView, StyleSheet, ActivityIndicator, Text, } from 'react-native'
import SyntaxHighlighter from 'react-native-syntax-highlighter'; 
import { dark} from 'react-syntax-highlighter/styles/hljs';
import HTMLView from 'react-native-htmlview';
import { useDispatch, useSelector } from 'react-redux';
import * as action from '../../actions/homeActions'
import { showInterstitialAd } from '../../ui/common/admob/InterstitialAdAdmob';
import { chosenArticleSelector, splitedArticleSelector } from '../../selectors/CodeAreaSelector';

const CodeArea = () => {
  const splitedArticle = useSelector(splitedArticleSelector)
  const chosenArticle = useSelector(chosenArticleSelector)

  const dispatch = useDispatch()

  useEffect(() => {
   
       dispatch(action.passSplitedMainTopicArticle(chosenArticle))
    return () => {
      dispatch(action.passSplitedMainTopicArticle(''))//reset article length in order to show actionindicator next time this screen loads
      // showInterstitialAd()
    }
  }, [])

  const splitBetweenTextAndCode= useCallback(() => {
      const elementsArr:any[]=[]

      Array.from(splitedArticle).forEach((textOrCode,index)=>{
   
       if(index%2!==0){
        elementsArr.push(
          <SyntaxHighlighter
            key={index}
            style={dark}
            customStyle={{}}
            language='kotlin'
            fontSize={15}
            highlighter={"hljs"}
          >
            {textOrCode}
          </SyntaxHighlighter>
        )
       }
       else{
        elementsArr.push(
          <View
          key={index} 
          style={{margin:10}}>
            <HTMLView
              value={`<p>${textOrCode}</p>`}
              stylesheet={htmlStyles}
            />
          </View>
        )
       }
      })
      return elementsArr
    },
    [splitedArticle]
  )

  return (
    
    <View style={{flex:1,backgroundColor:'#e6e6e6',flexDirection:'column'}}>
      {splitedArticle.length==0
        ? (<ActivityIndicator style={styles.indicator}  size='large' color={'#000'}/>)
        : (<ScrollView>{splitBetweenTextAndCode()}</ScrollView> )     
      }
    </View>
  )
}

const styles=StyleSheet.create({
  indicator:{
    marginTop:'auto',
    marginBottom:'auto',
  }
})

const htmlStyles=StyleSheet.create({
  p:{
    fontSize:15,
    fontFamily:'serif',
    lineHeight:25
  }
})

export default CodeArea
