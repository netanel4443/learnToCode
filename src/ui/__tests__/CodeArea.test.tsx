import {render} from '@testing-library/react-native'
import renderer from 'react-test-renderer'
import CodeArea from '../screens/CodeArea'
import React from 'react'
import * as redux from 'react-redux'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const mockStore=configureStore()

const store=mockStore()

const spy=jest.spyOn(redux,'useSelector')
spy.mockReturnValue({splitedArticle:['test','rferf']})
spy.mockReturnValue({chosenArticle:'Toast'})


const wrapper=renderer.create(<Provider store={store}><CodeArea/></Provider>).toJSON()

test('if renders correctly',(cb)=>{
  expect(wrapper).toMatchSnapshot();
  
})
 
