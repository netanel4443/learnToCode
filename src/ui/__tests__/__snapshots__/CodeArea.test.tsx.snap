// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e6e6e6",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <RCTScrollView>
    <View />
  </RCTScrollView>
</View>
`;
